samples/contrib/azure-samples/kfp-azure-databricks/databricks/_cluster_op.py

```diff
@@ -0,0 +1,274 @@
+                If no k8s_name is provided, cluster_name will be used as the resource name.
+                This name is DNS-1123 subdomain name and must consist of lower case alphanumeric
+                characters, '-' or '.', and must start and end with an alphanumeric character.
+            cluster_name: Cluster name requested by the user.
+            spec: Full specification of the Databricks cluster to create.
+            num_workers: Number of worker nodes that this cluster should have.
+            autoscale: Parameters needed in order to automatically scale clusters up and down based
+                on load.
+            spark_version: The runtime version of the cluster.
+            spark_conf: An object containing a set of optional, user-specified Spark configuration
+                key-value pairs.
+            node_type_id: This field encodes, through a single value, the resources available to
+                each of the Spark nodes in this cluster.
+            driver_node_type_id: The node type of the Spark driver.
+            custom_tags: Additional tags for cluster resources.
+            cluster_log_conf: The configuration for delivering Spark logs to a long-term storage
+                destination.
+            init_scripts: The configuration for storing init scripts.
+            spark_env_vars: An object containing a set of optional, user-specified environment
+                variable key-value pairs.
+            autotermination_minutes: Automatically terminates the cluster after it is inactive for
+                this time in minutes. If not set, this cluster will not be automatically terminated.
+            instance_pool_id: The optional ID of the instance pool to which the cluster belongs.
+
+        Raises:
+
+            ValueError: If no k8s resource name or Cluster name are provided.
+        """
+
+        if not spec:
+            spec = {}
+
+        if cluster_name:
+            spec["cluster_name"] = cluster_name
+        if num_workers:
+            spec["num_workers"] = num_workers
+        if autoscale:
+            spec["autoscale"] = autoscale
+        if spark_version:
+            spec["spark_version"] = spark_version
+        if spark_conf:
+            spec["spark_conf"] = spark_conf
+        if node_type_id:
+            spec["node_type_id"] = node_type_id
+        if driver_node_type_id:
+            spec["driver_node_type_id"] = driver_node_type_id
+        if custom_tags:
+            spec["custom_tags"] = custom_tags
+        if cluster_log_conf:
+            spec["cluster_log_conf"] = cluster_log_conf
+        if init_scripts:
+            spec["init_scripts"] = init_scripts
+        if spark_env_vars:
+            spec["spark_env_vars"] = spark_env_vars
+        if autotermination_minutes:
+            spec["autotermination_minutes"] = autotermination_minutes
+        if instance_pool_id:
+            spec["instance_pool_id"] = instance_pool_id
```

Comment:
Would it be possible to assign `spec["instance_pool_id"] = instance_pool_id` unconditionally?

Reply:
No, I don't think so. instance_pool_id is an optional field in the spec. It would appear in the spec with a value of None when it shouldn't.

Reply:
Thanks for the explanation.